src/ql/ir/ops.cc

```diff
@@ -343,6 +343,7 @@ InstructionTypeLink find_instruction_type(
 
     // If we shouldn't generate an overload if only the name matches, stop now.
```

Comment:
Just a note: I don't like the use of the term overload for an instruction type / prototype.
An overload would mean to me a companion instruction (or set of those) that also matches the instruction type, those overload each other.
Do you also see this?
Then I'd like to update the documentation/sources in this direction, once this review has completed.

Reply:
I'd say that the term 'overload' is used correctly here: you can have several instruction definitions with the same name, but different prototypes.

Reply:
I'm still studying on the relations between all of the following:

- overload and instruction type
- generalization and specialization
- which legacy decomposition rules apply
- prototypes and operand types

in the context of old to new conversion of a platform configuration file.
See also some of the other conversations.